llmkernel/kernel.py

```diff
@@ -32,107 +61,281 @@ class PythonLLMKernel(IPythonKernel):
         "file_extension": ".txt",
     }
 
+    internal_executions: set[str]
+    subkernel_execution_tracking: dict[str, str]
+
+    def __init__(self, server):
+        self.internal_executions = set()
+        self.subkernel_execution_tracking = {}
+        self.subkernel_id = None
+        super().__init__(server)
+        self.new_kernel()
+        self.setup_instance()
+
+    def setup_instance(self):
+        self.setup_llm()
+        self.setup_intercepts()
 
-    def setup_instance(self, *args, **kwargs):
-        self.toolset = None
-        self.agent = None
+    def setup_llm(self):
         # Init LLM agent
```

Comment:
How is this initializing? Isn't this a reset?

Reply:
You're right. There was more "initialization" in here previously at some point in the refactor, but that got moved to the set_context() function. I'll do a pass to cleanup.